app/models/piece.rb

```
@@ -32,4 +32,57 @@ def valid_move?
    
   end
 
+  def is_obstructed?(row_final, col_final)
+    row_init = self.position_row
+    col_init = self.position_column
+
+    def array_up(init, final)
+      array = (init...final).to_a
+      array = array[1,array.length]
+    end
+
+    def array_down(init, final)
+      array = init.downto(final).to_a
+      array = array[1, array.length-2]
+    end
+
+    #horizonal case
+    if row_init == row_final
+      if col_init <= col_final #going right
+        col_rng = array_up(col_init, col_final)
+      else #going left
+        col_rng = array_down(col_init, col_final)
+      end
+      return !Piece.exists?(position_row: row_init, position_column: col_rng)
```

@Edwin-F  I think @xelzi is right, when we call is_obstructed, the method should return true if there is a piece in the way, should return false if a piece is not in the way, and should raise an error if the piece doesn't move horizontal, vertical, or diagonal (i.e., a knight)